### PR TITLE
Providing data for Azure virtual machines

### PR DESCRIPTION
**Pull request by bpetit**
Hi there,

Here is a PR aggregating data to feed the API with Microsoft Azure data for virtual machines (and supposed underlying hosts).

This should not be merged until we at least clean it up (for instance, our scripts are **not PEP8 at all** 😋) and improve the readme (+ that it is reviewed of course), but we wanted to allow contributors to review it in advance and give feedbacks.

I hope it is not too much noise until it is clean.

kudos @bdromard and ping to join the discussion :)

### CODE REVIEW

File: boaviztapi/data/utils/complete_Azure_VMs/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- Bursts and overcommit are not accounted for, as the scripts in their current form only extract the number of vCPUs for each instance, without considering if this instance is a burstable instance. Fixing this would mean to extract informations about burst conditions for each instance and change the value of vcpus accordingly (or rely on a mecanism in BoaviztAPI that doesn't exist today).
+- For GPUs (concerning NVv3 and NVv4 series): GPU units and memory for the hosts are established by matching the number of vCPUS for an instance and the number of vCPUS for the host, and establishing
+the potential number of GPUs in the host with the documentation for the instances.
+- Current version of the API doesn't account for remote storage. A share of the impacts of local hardware storage is allocated to the virtual machine, which makes sense. In the case of Azure data, we don't have the actual stoarage hardware and volume available for Dedicated Hosts / Bare metal machines. It means that current version of the impacts calculated by the API for Azure is **underestimated** regarding impacts of storage, as we don't account neither for remote storage nor for local storage.
```
Comment (bpetit) [inline boaviztapi/data/utils/complete_Azure_VMs/README.md:21]: This looks like a modeling discussion we might have in the next Boavizta Tools meeting, isn't it ? @demeringo @da-ekchajzer

**Comment by bpetit** [comment]
Tried to explain a bit more the perimeter covered by the api's cloud/instance route [here](https://github.com/hubblo-org/boaviztapi/blob/dev/docs/docs/Explanations/services/cloud.md#perimeter-covered). I'd be happy to have your feedback on this :)

**Comment by bpetit** [comment]
Hi @da-ekchajzer @demeringo,

Data generated in result.csv seem coherent, not too far from data generated for AWS instances with the same script.

I think we could procede with a global review any time you want now.

Last thing I may do, but it could be in another PR, will be to include data for instances still missing, especially A and B series.

As said in the README, we thank [GreenPixie](https://greenpixie.com/) for sponsoring this PR.

**Comment by demeringo** [comment]
> Tried to explain a bit more the perimeter covered by the api's cloud/instance route [here](https://github.com/hubblo-org/boaviztapi/blob/dev/docs/docs/Explanations/services/cloud.md#perimeter-covered). I'd be happy to have your feedback on this :)

Great, particularly the diagram 👍

**Comment by havontuur** [comment]
@bpetit I have looked and tried the PR and might have found a misconfiguration for 'id=Dadsv5-Type1' in server.csv.
In this PR this platform is defined with 112 vCPU and a SSD of 2400, but according the Microsoft page it should be 96 and 3600. 

Am I right or do I misunderstand the configuration?

**Comment by bdromard** [comment]
> @bpetit I have looked and tried the PR and might have found a misconfiguration for 'id=Dadsv5-Type1' in server.csv. In this PR this platform is defined with 112 vCPU and a SSD of 2400, but according the Microsoft page it should be 96 and 3600.
> 
> Am I right or do I misunderstand the configuration?

Hello @havontuur !

What is indicated for `Dadsv5-Type1` in `server.csv` (or any other Azure ID in that table) is the configuration for the dedicated host identified with this name, as documented on [this page](https://learn.microsoft.com/en-us/azure/virtual-machines/dedicated-host-general-purpose-skus) for the vCPU and the RAM specifications.

On the issue of storage, the allocation remains a problem, as we have not found reliable information on the hardware devices present on dedicated hosts yet.

Were you referring to [this page](https://learn.microsoft.com/en-us/azure/virtual-machines/dasv5-dadsv5-series#dadsv5-series) from the Azure documentation on virtual machines when you were mentioning 96 vCPU and 3600 GiB of SSD storage ?

**Pull request merged**